src/internal/m365/onedrive/export_test.go

```diff
@@ -0,0 +1,401 @@
+					},
+				},
+			},
+		},
+		{
+			name:    "multiple items",
+			version: 1,
+			backingCollections: []data.RestoreCollection{
+				data.NoFetchRestoreCollection{
+					Collection: mockRestoreCollection{
+						items: []data.Stream{
+							mockDataStream{id: "name1", data: "body1"},
+							mockDataStream{id: "name2", data: "body2"},
+						},
+					},
+				},
+			},
+			expectedItems: []data.ExportItem{
+				{
+					ID: "name1",
+					Data: data.ExportItemData{
+						Name: "name1",
+						Body: io.NopCloser((bytes.NewBufferString("body1"))),
+					},
+				},
+				{
+					ID: "name2",
+					Data: data.ExportItemData{
+						Name: "name2",
+						Body: io.NopCloser((bytes.NewBufferString("body2"))),
+					},
+				},
+			},
+		},
+		{
+			name:    "single item with data suffix",
+			version: 2,
+			backingCollections: []data.RestoreCollection{
+				data.NoFetchRestoreCollection{
+					Collection: mockRestoreCollection{
+						items: []data.Stream{
+							mockDataStream{id: "name1.data", data: "body1"},
+						},
+					},
+				},
+			},
+			expectedItems: []data.ExportItem{
+				{
+					ID: "name1.data",
+					Data: data.ExportItemData{
+						Name: "name1",
+						Body: io.NopCloser((bytes.NewBufferString("body1"))),
+					},
+				},
+			},
+		},
+		{
+			name:    "single item name from metadata",
+			version: version.Backup,
+			backingCollections: []data.RestoreCollection{
+				data.FetchRestoreCollection{
+					Collection: mockRestoreCollection{
+						items: []data.Stream{
+							mockDataStream{id: "id1.data", data: "body1"},
+						},
+					},
+					FetchItemByNamer: finD{id: "id1.data", name: "name1"},
+				},
+			},
+			expectedItems: []data.ExportItem{
+				{
+					ID: "id1.data",
+					Data: data.ExportItemData{
+						Name: "name1",
+						Body: io.NopCloser((bytes.NewBufferString("body1"))),
+					},
+				},
+			},
+		},
+		{
+			name:    "single item name from metadata with error",
+			version: version.Backup,
+			backingCollections: []data.RestoreCollection{
+				data.FetchRestoreCollection{
+					Collection: mockRestoreCollection{
+						items: []data.Stream{
+							mockDataStream{id: "id1.data"},
+						},
+					},
+					FetchItemByNamer: finD{err: assert.AnError},
+				},
+			},
+			expectedItems: []data.ExportItem{
+				{
+					ID:    "id1.data",
+					Error: assert.AnError,
+				},
+			},
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			ec := exportCollection{
+				baseDir:            "",
+				backingCollections: test.backingCollections,
+				version:            test.version,
+			}
+
+			items := ec.GetItems(ctx)
+
+			fitems := []data.ExportItem{}
+			for item := range items {
+				fitems = append(fitems, item)
+			}
+
+			assert.Len(t, fitems, len(test.expectedItems), "num of items")
+
+			for i, item := range fitems {
+				assert.Equal(t, test.expectedItems[i].ID, item.ID, "id")
```

Comment:
I'm a bit on the fence about expecting things in the same order. Nothing actually depends on the returned order of the items (and really we don't want to enforce some ordering in the exported API if we don't have to). Unfortunately, not relying on ordering seems like it would just make the test more complex since we need non-trivial comparisons for things like errors

Reply:
I've left a note in there for now. We can update it later if necessary I guess.